### PR TITLE
Fixes misprints, duplicated value assignments, possible usage of null pointers

### PR DESCRIPTION
**Pull request by qarmin**
Most of changes remove duplicated assignments value to variables:
```
var = 1
...
... Code which not using `var`
...
var = 1
```
In `rvo_agent.cpp` and `display_server_x11.cpp` the pointers were used without checking if they are `nullptr` but in other parts of code were checked.

### CODE REVIEW

File: modules/gdnavigation/rvo_agent.cpp [hunk 1]
```diff
@@ -80,5 +80,7 @@ void RvoAgent::dispatch_callback() {
 
 	const Variant *vp[2] = { &callback.new_velocity, &callback.udata };
 	int argc = (callback.udata.get_type() == Variant::NIL) ? 1 : 2;
-	obj->call(callback.method, vp, argc, responseCallError);
+	if (obj) {
```
Comment (akien-mga) [inline modules/gdnavigation/rvo_agent.cpp:83]: I think this should be an early return in the previous `if (obj == nullptr)` instead.
@AndreaCatania

File: platform/linuxbsd/display_server_x11.cpp [hunk 2]
```diff
@@ -3806,7 +3806,9 @@ DisplayServerX11::~DisplayServerX11() {
 	for (Map<WindowID, WindowData>::Element *E = windows.front(); E; E = E->next()) {
 #ifdef VULKAN_ENABLED
 		if (rendering_driver == "vulkan") {
-			context_vulkan->window_destroy(E->key());
+			if (context_vulkan) {
```
Comment (akien-mga) [inline platform/linuxbsd/display_server_x11.cpp:3809]: I'd make this an `ERR_FAIL_NULL`, if we don't have a context to destroy the windows, but we still have windows to destroy, we're going to leak (or crash).

File: servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp [hunk 3]
```diff
@@ -1536,7 +1536,6 @@ RasterizerEffectsRD::RasterizerEffectsRD() {
 
 	default_sampler = RD::get_singleton()->sampler_create(sampler);
 
-	sampler.min_filter = RD::SAMPLER_FILTER_LINEAR;
```
Comment (akien-mga) [inline servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp:1539]: CC @clayjohn, looks a bit weird that not all members are set for each `sampler`.
Logic is preserved by the change though.

Comment (clayjohn) [reply servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp:1539]: Change seems fine to me, reduz does the same thing in other places anyway.

File: modules/gdnavigation/rvo_agent.cpp (hunk 1 above)
Comment (AndreaCatania) [reply modules/gdnavigation/rvo_agent.cpp:82]: Yes, it should be an early return on the above `if`.

File: modules/gdnavigation/rvo_agent.cpp [hunk 4]
```diff
@@ -70,7 +70,7 @@ void RvoAgent::dispatch_callback() {
 	}
 	Object *obj = ObjectDB::get_instance(callback.id);
 	if (obj == nullptr) {
-		callback.id = ObjectID();
+		return;
```
Comment (AndreaCatania) [inline modules/gdnavigation/rvo_agent.cpp:73]: Return here is fine, but you also need `callback.id = ObjectID();` to reset the id. In this way the next frame, the above check pass and it returns earlier.

File: servers/rendering/renderer_rd/cluster_builder_rd.h [hunk 5]
```diff
@@ -235,7 +235,7 @@ class ClusterBuilderRD {
 		Transform xform = view_xform * p_transform;
 
 		float radius = xform.basis.get_uniform_scale();
-		if (radius > 0.98 || radius < 1.02) {
+		if (radius > 0.98 && radius < 1.02) {
```
Comment (qarmin) [inline servers/rendering/renderer_rd/cluster_builder_rd.h:238]: With this one, I'm not sure.
Should be `(radius > 0.98 && radius < 1.02)` or maybe `(radius < 0.98 || radius > 1.02)`

Comment (clayjohn) [reply servers/rendering/renderer_rd/cluster_builder_rd.h:238]: We will need to check with @reduz

**Comment by qarmin** [comment]
Most things was fixed, rest is discussed in different issues/PRs

**Pull request closed without merging**